sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+        [propertyName: string]: string;
+    };
+    // (undocumented)
+    driverCores?: number;
+    // (undocumented)
+    driverMemory?: string;
+    // (undocumented)
+    executorCores?: number;
+    // (undocumented)
+    executorCount?: number;
+    // (undocumented)
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
+    // (undocumented)
+    files?: string[];
+    // (undocumented)
+    jars?: string[];
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    pythonFiles?: string[];
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export type SparkBatchJobResultType = "Uncertain" | "Succeeded" | "Failed" | "Cancelled" | string;
+
+// @public (undocumented)
+export interface SparkBatchJobState {
+    currentState?: string;
+    deadAt?: Date;
+    // (undocumented)
+    jobCreationRequest?: SparkRequest;
+    notStartedAt?: Date;
+    recoveringAt?: Date;
+    runningAt?: Date;
+    startingAt?: Date;
+    successAt?: Date;
+    terminatedAt?: Date;
+}
+
+// @public (undocumented)
+export class SparkClient {
+    constructor(workspaceEndpoint: string, sparkPoolName: string, credential: TokenCredential, pipelineOptions?: SparkClientOptions);
+    cancelSparkBatchJob(batchId: number, options?: CancelSparkBatchJobOptions): Promise<OperationResponse>;
+    cancelSparkSession(sessionId: number, options?: CancelSparkSessionOptions): Promise<OperationResponse>;
+    cancelSparkStatement(sessionId: number, statementId: number, options?: CancelSparkStatementOptions): Promise<OperationResponse>;
+    createSparkBatchJob(sparkBatchJobOptions: SparkBatchJobOptions, options?: CreateSparkBatchJobOptions): Promise<CreateSparkBatchJobResponse>;
+    createSparkSeesion(sparkSessionOptions: SparkSessionOptions, options?: CreateSparkSessionOptions): Promise<CreateSparkSessionResponse>;
+    createSparkStatement(sessionId: number, sparkStatementOptions: SparkStatementOptions, options?: CreateSparkStatementOptions): Promise<CreateSparkStatementResponse>;
+    getSparkBatchJob(batchId: number, options?: GetSparkBatchJobOptions): Promise<GetSparkBatchJobResponse>;
+    getSparkSession(sessionId: number, options?: GetSparkSessionOptions): Promise<GetSparkSessionResponse>;
+    getSparkStatement(sessionId: number, statementId: number, options?: GetSparkStatementOptions): Promise<GetSparkStatementResponse>;
+    listSparkBatchJobs(options?: ListSparkBatchJobsOptions): Promise<ListSparkBatchJobsResponse>;
+    listSparkSessions(options?: ListSparkSessionsOptions): Promise<ListSparkSessionsResponse>;
+    listSparkStatements(sessionId: number, options?: ListSparkStatementsOptions): Promise<ListSparkStatementsResponse>;
+    resetSparkSessionTimeout(sessionId: number, options: ResetSparkSessionTimeoutOptions): Promise<OperationResponse>;
+    readonly workspaceEndpoint: string;
+}
+
+// @public
+export interface SparkClientOptions extends PipelineOptions {
+    endpoint?: string;
```

Comment:
It is provider a way to override baseurl. We can find its logic at [here](https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/synapse/synapse-spark/src/generated/synapseSparkContext.ts#L50).

In fact, that part is generated by Autorest, @idear1203 could share some context about it, because it exits all language SDK, but we don't seem to need it

Reply:
Useless,. will removed that.

---
In reply to: [521600916](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521600916) [](ancestors = 521600916)